openquake/hazardlib/near_fault.py

```diff
@@ -0,0 +1,476 @@
+    [site_x, site_y, site_z] = get_xyz_from_ll(site, reference)
+
+    a = np.array([(1, 0, 0, -normal[0]),
+                  (0, 1, 0, -normal[1]),
+                  (0, 0, 1, -normal[2]),
+                  (normal[0], normal[1], normal[2], 0)])
+    b = np.array([site_x, site_y, site_z, dist_to_plane])
+
+    x = np.linalg.solve(a, b)
+    pp = np.array([x[0], x[1], x[2]])
+    return pp
+
+
+def vectors2angle(v1, v2):
+    """ Returns the angle in radians between vectors 'v1' and 'v2'.
+
+    :param v1:
+        vector, format in numpy array
+    :param v2:
+        vector, format in numpy array
+    :return:
+        the angle in radians between the two vetors
+    """
+    cosang = np.dot(v1, v2)
+    sinang = np.linalg.norm(np.cross(v1, v2))
+    return np.arctan2(sinang, cosang)
+
+
+def average_s_rad(site, hypocenter, reference, pp,
+                  normal, dist_to_plane, e, p0, p1, delta_slip):
+    """
+    Get the average S-wave radiation pattern over E-path desrcibed in:
+    "Final report of the NGA-West2 directivity working group." (2013) by
+    Spudich, Paul, et al. and publish in PEER report, page 90- 92. Also
+    obtain the site to direct point distance, rd, and the hypocentre
+    distance, r_hyp.
+
+    :param site:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the location of the target site
+    :param hypocenter:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the location of hypocenter
+    :param reference:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the location
+        of the reference point for coordinate projection within the
+        calculation. The suggested reference point is Epicentre.
+    :param pp:
+        the projection point pp on the patch plane,
+        format in numpy array
+    :param normal:
+        normal of the plane, describe by a normal vector[a, b, c]
+    :param dist_to_plane:
+        d in the plane equation,  ax + by + cz = d
+    :param e:
+        E-path, in km.
+    :param p0:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the location of the starting point on fault segment
+    :param p1:
+        :class:`~openquake.hazardlib.geo.point.Point` object
+        representing the location of the ending point on fault segment.
+    :param delta_slip:
+        slip direction away from the strike direction, in decimal degrees.
+        A positive angle is generated by a counter-clockwise rotation.
+    :return:
+        fs, float value of the average S-wave radiation pattern.
+        rd, the distance from site to the direct point.
+        r_hyp, the hypocetre distance. Both distance calculation has been
+        sugggested in the Spudich, et al.(2013)
+    """
+    # zs
+
+    site_xyz = get_xyz_from_ll(site, reference)
+    zs = dst.pdist([pp, site_xyz])
+
+    if site_xyz[0] * normal[0] + site_xyz[1] * normal[1] + site_xyz[2] * \
```

Comment:
In the PEER report, Spudich et. al., 2013 page 91-92
Zs is a signed distance between PS and PP
Variable Zs the 𝑛- coordinate of Ps, is positive when Ps(site) is above the fault plane(the target fault patch).
Here, I want to define if the site is above the fault plan or not.